src/packages/settings/dashboards/performance-profiling/dashboard-performance-profiling.element.ts

```diff
@@ -1,88 +1,68 @@
-import { UmbTextStyles } from "@umbraco-cms/backoffice/style";
-import { css, html, customElement, state } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { css, html, customElement, state, query, unsafeHTML } from '@umbraco-cms/backoffice/external/lit';
 import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
 import { ProfilingResource } from '@umbraco-cms/backoffice/backend-api';
 import { tryExecuteAndNotify } from '@umbraco-cms/backoffice/resources';
 
 @customElement('umb-dashboard-performance-profiling')
 export class UmbDashboardPerformanceProfilingElement extends UmbLitElement {
 	@state()
-	private _profilingStatus?: boolean;
+	private _profilingStatus = true;
 
 	// TODO: Get this from the management api configuration when available
 	@state()
 	private _isDebugMode = true;
 
+	@query('#toggle')
+	private _toggle!: HTMLInputElement;
+
+	#setToggle(value: boolean) {
+		this._toggle.checked = value;
+		this._profilingStatus = value;
+	}
+
 	firstUpdated() {
 		this._getProfilingStatus();
 	}
 
 	private async _getProfilingStatus() {
 		const { data } = await tryExecuteAndNotify(this, ProfilingResource.getProfilingStatus());
 
-		if (data) {
-			this._profilingStatus = data.enabled;
-		}
+		if (!data || !data.enabled) return;
+		this._profilingStatus = data.enabled;
 	}
 
 	private async _changeProfilingStatus() {
 		const { error } = await tryExecuteAndNotify(
 			this,
-			ProfilingResource.putProfilingStatus({ requestBody: { enabled: !this._profilingStatus } })
+			ProfilingResource.putProfilingStatus({ requestBody: { enabled: !this._profilingStatus } }),
 		);
 
-		if (!error) {
-			this._profilingStatus = !this._profilingStatus;
-		}
+		error ? this.#setToggle(this._profilingStatus) : this.#setToggle(!this._profilingStatus);
```

Comment:
Please use the `if (error) {}` notation. It makes the code easier to read ☺️

Reply:
done :)